esi_leap/send_email_notification.py

```diff
@@ -0,0 +1,173 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+"""
+Description:
+This script queries and sends email notifications for ESI lease events.
+
+Usage:
+Before running, ensure the following environment variables are set:
+- ENABLE_EMAIL: Set to 'true' to enable email notifications.
+  Default is 'false'.
+- SMTP_SERVER: Hostname of the SMTP server for sending emails. Default is
+  'localhost'
+- EMAIL_SENDER: The email address used as the sender in outgoing emails
+- LEASE_WARNING_DAYS: Days before a lease's end to send a notification email
+- LEASE_CREATE_TEMPLATE: Path to the email template for lease creation
+  notifications. Default is esi_leap/templates/lease_create_email.txt
+- LEASE_EXPIRE_TEMPLATE: Path to the email template for lease expiration
+  warnings. Default is esi_leap/templates/lease_expire_email.txt
+
+After esi-leap package is installed, run `esi_leap_query_events`
```

Comment:
Would you run `esi-leap-email-notification` instead? I might mention that the script would be run periodically as well.

Reply:
The script itself doesn't have scheduling mechanism. But we can make it a cronjob to run periodically. Just add the command `esi-leap-email-notification` in a crontab file and specify a frequency would work.

Reply:
Yep, sorry, I was unclear - the documentation should be updated to say "run esi-leap-email-notification" instead of "esi_leap_query_events", right? And I think the documentation should tell the operator to set up a periodic cron job rather than just running it once.

Reply:
Makes sense! I just added this instruction in the documentation.